src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2123,6 +2156,123 @@ private void LoadEnvironmentVariable(ProcessStartInfo startinfo, IDictionary Env
             }
         }
 
+        /// <summary>
+        /// Attempt to stop the process when the timeout has expired.
+        /// <see cref="StopProcessCommand" /> is used to stop the process
+        /// </summary>
+        /// <param name="process">
+        /// The process that should be stopped
+        /// </param>
+        private void StopProcessOnTimeout(Process process)
+        {
+            string message = StringUtil.Format(ProcessResources.StartProcessTimeoutExceeded, process.ProcessName);
+            ErrorRecord er = new ErrorRecord(new TimeoutException(message), "StartProcessTimeoutExceeded", ErrorCategory.OperationTimeout, process);
+
+            StopProcessCommand stop = new StopProcessCommand();
```

Comment:
Seems like Stop-Process should have a switch called `-IncludeChildProcesses` and you would use that here

Reply:
Ideally. As there currently isn't one, I implemented the `GetProcessTreeIds` function to do this role.

If this code is accepted, I had planned to open an issue / PR about somehow integrating this function into the `Stop-Process` cmdlet in the future (so as not to change too much code in this PR), instead of it being located in `Start-Process`.